### PR TITLE
ID5 User ID module - don't send empty fields to server

### PR DESCRIPTION
**Pull request by smenzer**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
For optional fields, if they are empty don't send them to the server. Also add some logging to help with debugging

### CODE REVIEW

File: test/spec/modules/id5IdSystem_spec.js
Comment (Fawke) [inline test/spec/modules/id5IdSystem_spec.js:22]: I think you forgot to get rid of `.only`, because of this all test cases are not running.

**Review by Fawke — CHANGES_REQUESTED** [review]
Hi @smenzer,

The change looks good. Can you just make sure that all test cases run properly by removing .only from your test case file.

As of now, I see few failures when I run `gulp test`.

Screenshot 2021-04-14 at 1 35 08 PM

File: test/spec/modules/id5IdSystem_spec.js
Comment (smenzer) [reply test/spec/modules/id5IdSystem_spec.js:22]: whoops my bad...fixed now

**Comment by smenzer** [comment]
> As of now, I see few failures when I run `gulp test`.

Yea my bad...I added some `logInfo()`'s after finishing my code and forgot to run tests again. should be working now, @Fawke

**Review by Fawke — APPROVED** [review]
LGTM

**Comment by smenzer** [comment]
@Fawke any reason in particular this needs a second review? it's not core...

**Comment by smenzer** [comment]
@Fawke or @ChrisHuie just wanted to check if this really needs a 2nd review, and if not, can we merge it?

**Comment by Fawke** [comment]
@smenzer I guess we can merge it, sorry for the confusion. I read the PR Review guidelines and there's this clause:

> If all above is good, add a LGTM comment and, if the change is in PBS-core or is an important module like the prebidServerBidAdapter, request 1 additional core member to review.

It's kind of a grey area what constitutes an important module, but I think, this change's okay, since it's mostly minor enhancements and refactoring.

**Comment by smenzer** [comment]
> It's kind of a grey area what constitutes an important module, but I think, this change's okay, since it's mostly minor enhancements and refactoring.

got it, thanks for pushing through, @Fawke

**Pull request merged**
